Source/Base/ASDisplayNode+Ancestry.m

```diff
@@ -0,0 +1,54 @@
+//
+//  ASNodeAncestorEnumerator.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 4/12/17.
+//  Copyright © 2017 Facebook. All rights reserved.
+//
+
+#import "ASDisplayNode+Ancestry.h"
+
+AS_SUBCLASSING_RESTRICTED
+@interface ASNodeAncestryEnumerator : NSEnumerator
+@end
+
+@implementation ASNodeAncestryEnumerator {
+  // Enumerators are one-shot optimized – avoid retains/releases/weak
+  __unsafe_unretained ASDisplayNode * _Nullable _node;
```

Comment:
Hmmm under normal circumstances I would say yes, but since we allow you to modify node hierarchies from any thread, I guess we have to use `weak` here, since the "collection" we are enumerating can't be held still.